server/rooms.js

```diff
@@ -591,37 +591,55 @@ class GlobalRoom extends BasicRoom {
 		}
 		return this.formatList;
 	}
-	get configRankList() {
-		if (Config.nocustomgrouplist) return '';
-
-		// putting the resultant object in Config would enable this to be run again should config.js be reloaded.
-		if (Config.rankList) {
-			return Config.rankList;
-		}
-		let rankList = [];
-
-		for (let rank in Config.groups) {
-			if (!Config.groups[rank] || !rank) continue;
-
-			let tarGroup = Config.groups[rank];
-			let groupType = tarGroup.addhtml || (!tarGroup.mute && !tarGroup.root) ? 'normal' : (tarGroup.root || tarGroup.declare) ? 'leadership' : 'staff';
-
-			rankList.push({symbol: rank, name: (Config.groups[rank].name || null), type: groupType}); // send the first character in the rank, incase they put a string several characters long
+	/**
+	 * @param {AnyObject} group
```

Comment:
Is this really `AnyObject`? You seem to depend on `symbol`, `id` and `name` keys  - that should be reflected here

Reply:
`group` here is the group as it's defined in config, not the group object with `symbol`, `id`, `name`, and `type` keys

Reply:
Wait, I misread. How would I type it so it includes those three keys but ignores any others the group may have?

Reply:
Thats how interfaces work already with structural typing? 

```ts
interface Foo {
  foo: number;
}

function inc(foo: Foo) {
  return foo.foo + 1;
}

const bar = {foo: 5, bar: 10}; // bar implicitly implements Foo
console.log(inc(bar)); // => 6
```

Reply:
Though actually, looking at it again, you care about more than just `symbol`, `id`, and `name` (you want `editroom`, `makeroom`, `root` etc).